compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.desktop.kt

```diff
@@ -93,21 +92,50 @@ internal class ComposeSceneMediator(
         private val clipMap = mutableMapOf<Component, ClipComponent>()
 
         override fun componentAdded(e: ContainerEvent) {
+            val component = e.child
             if (useInteropBlending) {
-                return
+                // In case of interop blending, compose might draw content above this [component].
+                // But due to implementation of [JLayeredPane]'s lightweight/heavyweight mixing
+                // logic, it doesn't send mouse events to parents or another layers.
+                // In case if [component] is placed above [contentComponent] (see addToLayer),
+                // subscribe to mouse events from interop views to handle such input.
+                subscribeToMouseEvents(component)
+            } else {
+                // Without interop blending, just add clip region to make proper
+                // "interop always on top" behaviour.
+                addClipComponent(component)
             }
+        }
+
+        override fun componentRemoved(e: ContainerEvent) {
             val component = e.child
+            removeClipComponent(component)
+            unsubscribeFromMouseEvents(component)
+        }
+
+        private fun addClipComponent(component: Component) {
             val clipRectangle = ClipComponent(component)
             clipMap[component] = clipRectangle
             skiaLayerComponent.clipComponents.add(clipRectangle)
         }
 
-        override fun componentRemoved(e: ContainerEvent) {
-            val component = e.child
+        private fun removeClipComponent(component: Component) {
             clipMap.remove(component)?.let {
                 skiaLayerComponent.clipComponents.remove(it)
             }
         }
+
+        private fun subscribeToMouseEvents(component: Component) {
+            component.addMouseListener(mouseListener)
+            component.addMouseMotionListener(mouseListener)
+            component.addMouseWheelListener(mouseListener)
+        }
+
+        private fun unsubscribeFromMouseEvents(component: Component) {
+            component.removeMouseListener(mouseListener)
+            component.removeMouseMotionListener(mouseListener)
+            component.removeMouseWheelListener(mouseListener)
+        }
```

Comment:
This
1. Can be in the enclosing class
2. Can be reused in `subscribe`
3. Can be `Component.subscribeToMouseEvents`.

Reply:
Done